webapp/components/Map/ShowZoomLevel.js

```diff
@@ -0,0 +1,26 @@
+import './ShowZoomLevel.scss'
+
+import React, { useEffect, useState } from 'react'
+import { useMap } from 'react-leaflet'
+
+import * as User from '@core/user/user'
+import { useUser } from '@webapp/store/user'
+
+export const ShowZoomLevel = () => {
+  const map = useMap()
+  const user = useUser()
+  const [zoomLevel, setZoomLevel] = useState()
+
+  useEffect(() => {
+    setZoomLevel(map.getZoom())
+  }, [])
+
+  map.on('zoomend', () => {
+    setZoomLevel(map.getZoom())
+  })
```

Comment:
do you need to remove this listener?

Reply:
What do you mean?

Reply:
Oh I think there is a confusion because of the name? Maybe should change that. setZoomLevel sets the number value on the component that is shown on top of a map, to show what is is the current zoom level. Its "set" because that is the standard way to name state hooks. It does not set the zoom of the map object.

Reply:
the map.on('event', is the listener, not sure if you need to remove the listener when you unmount the component

Reply:
Well it needs to live as long as the map lives, so unless there is a map destroyer somewhere, then I doubt

Reply:
https://www.pluralsight.com/guides/event-listeners-in-react-components

Reply:
You can useMapEvents or useMapEvent, they take the responsibility of mounting and unmounting the listeners

Also as you can see they unmount the listener with 'map.off' when the component is unmounted.
 https://github.com/PaulLeCam/react-leaflet/blob/1b6846de786c19ce70d8514c174b9bd020fd647f/packages/react-leaflet/src/hooks.ts#L15